azurelinuxagent/ga/update.py

```diff
@@ -94,26 +95,32 @@ def run_latest(self):
         """
         latest_agent = self.get_latest_agent()
         if latest_agent == None:
-            agent_bin = sys.argv[0]
+            agent_cmd = "python -u {0} -run-exthandlers".format(sys.argv[0])
```

Comment:
distros like ubuntu16.04 use /usr/bin/python3 explicitly to run the agent. this will use python2 to run the agent.

I still think having a shebang like `#!/usr/bin/env python -u` in the /sbin/waagent (or whatever) is a better option to have u at.